admin/database/postgres/postgres.go

```diff
@@ -2060,25 +2049,108 @@ func (c *connection) DeleteBillingIssueByTypeForOrg(ctx context.Context, orgID s
 	return checkDeleteRow("billing issue", res, err)
 }
 
-// projectDTO wraps database.Project, using the pgtype package to handle types that pgx can't read directly into their native Go types.
-type projectDTO struct {
-	*database.Project
-	ProdVariables pgtype.JSON `db:"prod_variables"`
-	Annotations   pgtype.JSON `db:"annotations"`
+func (c *connection) FindProjectVariablesByEnvironment(ctx context.Context, projectID string, environment *string) ([]*database.ProjectVariable, error) {
+	q := `SELECT * FROM project_variables WHERE project_id = $1`
+	args := []interface{}{projectID}
+	if environment != nil {
+		q += ` AND environment = $2`
+		args = append(args, environment)
+	}
+	var res []*database.ProjectVariable
+	err := c.getDB(ctx).SelectContext(ctx, &res, q, args...)
+	if err != nil {
+		return nil, parseErr("project variables", err)
+	}
+
+	// Decrypt the variables
+	for _, v := range res {
+		decryptedValue, err := c.decrypt(v.Value, v.ValueEncryptionKeyID)
+		if err != nil {
+			return nil, err
+		}
+		v.Value = decryptedValue
+	}
+
+	return res, nil
 }
 
-func (c *connection) projectFromDTO(dto *projectDTO) (*database.Project, error) {
-	err := dto.ProdVariables.AssignTo(&dto.Project.ProdVariables)
-	if err != nil {
-		return nil, err
+func (c *connection) UpsertProjectVariable(ctx context.Context, projectID, environment string, vars map[string]string, userID string) ([]*database.ProjectVariable, error) {
+	query := `INSERT INTO project_variables (project_id, environment, name, value, value_encryption_key_id, updated_by_user_id, updated_on)
+	VALUES (:project_id, :environment, :name, :value, :value_encryption_key_id, :updated_by_user_id, now())
+	ON CONFLICT (project_id, environment, lower(name)) DO UPDATE SET
+		value = EXCLUDED.value,
+		value_encryption_key_id = EXCLUDED.value_encryption_key_id,
+		updated_by_user_id = EXCLUDED.updated_by_user_id,
+		updated_on = now() RETURNING *`
+
+	args := make([]database.ProjectVariable, 0, len(vars))
```

Comment:
The column specified in the `INSERT` statement should ensure that only the required fields are set. We can also create a map since the args can also be a slice of maps. Just wanted to re-use the struct for simple use case.